Tests/Fakes/Song.swift

```diff
@@ -70,11 +64,28 @@ extension Song {
             }
         }
 
-        var rootKey: String? {
+        var rootKey: String {
             switch self {
             case .getEnvelopedSongs:
                 return "songs"
             }
         }
+
+        func serialize(response: Result<Data>) -> SerializationResult<[Song]> {
```

Comment:
One of Wolf's design objectives was to eliminate (most of) the need of manually implementing this `serialize(response:)` method. Can we achieve the same behavior with `Decodable`?

Reply:
I had some problems trying to match the extension with the `Decodable` alongside with `Collection`, but i've refactored the serialize code to achieve the expected result, check it out and tell me what you think